rush/rush-lib/src/data/VersionMismatchFinder.ts

```diff
@@ -0,0 +1,71 @@
+import RushConfigurationProject from './RushConfigurationProject';
+export class VersionMismatchFinder {
+  private _projects: RushConfigurationProject[];
+ /* store it like this:
+  * {
+  *   "@types/node": {
+  *     "1.0.0": [ '@ms/rush' ]
+  *   }
+  * }
+  */
+  private _mismatches: Map<string, Map<string, string[]>>;
+  constructor(projects: RushConfigurationProject[]) {
+    this._projects = projects;
+    this._mismatches = new Map<string, Map<string, string[]>>();
+    this._analyze();
+  }
+  public get numberOfMismatches(): number {
+    return this._mismatches.size;
+  }
+  public getMismatches(): Array<string> {
+    return this._iterableToArray<string>(this._mismatches.keys());
+  }
+  public getVersionsOfMismatch(mismatch: string): Array<string> {
+    return this._mismatches.has(mismatch)
+      ? this._iterableToArray<string>(this._mismatches.get(mismatch).keys())
+      : undefined;
+  }
+  public getConsumersOfMismatch(mismatch: string, version: string): Array<string> {
+    const mismatchedPackage: Map<string, string[]> = this._mismatches.get(mismatch);
+    if (!mismatchedPackage) {
+      return undefined;
+    }
+    const mismatchedVersion: string[] = mismatchedPackage.get(version);
+    return mismatchedVersion;
+  }
+  private _analyze(): void {
+    this._projects.forEach((project: RushConfigurationProject) => {
+      this._addDependenciesToList(project.packageName, project.packageJson.dependencies);
+      this._addDependenciesToList(project.packageName, project.packageJson.devDependencies);
+      // tslint:disable:no-any
+      this._addDependenciesToList(project.packageName, (project.packageJson as any).peerDependencies);
```

Comment:
Can we just add `peerDependencies` and `optionalDependencies` to the package.json typings?